src/transforms/at-range.js

```diff
@@ -113,14 +113,23 @@ Transforms.deleteAtRange = (transform, range, options = {}) => {
   const endBlock = document.getClosestBlock(document.getNextText(endKey).key)
 
   if (startBlock.key !== endBlock.key) {
-    endBlock.nodes.forEach((child, i) => {
-      const newKey = startBlock.key
-      const newIndex = startBlock.nodes.size + i
+    let blockToRemove = endBlock
+    let blockToKeep = startBlock
+
+    // We keep the style of the blockToKeep. endBlock should keep style if deleting empty line.
+    if (startBlock.isEmpty && !endBlock.isEmpty) {
+      blockToRemove = startBlock
+      blockToKeep = endBlock
+    }
+
+    blockToRemove.nodes.forEach((child, i) => {
+      const newKey = blockToKeep.key
+      const newIndex = blockToKeep.nodes.size + i
       transform.moveNodeByKey(child.key, newKey, newIndex, OPTS)
     })
 
-    // Remove parents of endBlock as long as they have a single child
-    const lonely = document.getFurthestOnlyChildAncestor(endBlock.key) || endBlock
+    // Remove parents of blockToRemove as long as they have a single child
+    const lonely = document.getFurthestOnlyChildAncestor(blockToRemove.key) || blockToRemove
```

Comment:
It retrieves the highest ancestor of `blockToRemove` that has only a single child, (in this case probably an empty text node, not sure).